### PR TITLE
US135815 Lit 2 Migration - Phase 2

### PR DESCRIPTION
**Pull request by jguann**
[US135815 ](https://rally1.rallydev.com/#/431372673576ud/iterationstatus?detail=%2Fuserstory%2F623960400869)

This PR makes the breaking changes required to update to Lit 2 (`lit-html` 2 and `lit-element` 3). The Lit 2 upgrade will be a coordinated effort, so **this PR should not merge at this time**. 

Breaking change checklist:  
- [x] Update `lit-html` -> `^2`, `lit-element` -> `^3`, `open-wc/testing` -> `^3` (if used) 
- [x] `createRenderRoot` usages -> move from LitElement to ReactiveElement  
  - No usages. 
- [x] Remove Dependabot Lit rules 
  - No usages. 
- [x] Bump any dependencies to their Lit 2 versions 
  - Core -> 2, htmleditor -> 2, view-toggle -> 1 
- [x] Remove any manual lifecycle calls to Lit 1 reactive controllers 
  - No usages. 
- [x] Backwards-compatible changes merged: https://github.com/Brightspace/continuous-professional-development/pull/389  

Renamed API's: (No usages) 
- [x] `UpdatingElement` -> `ReactiveElement` 
- [x] `@internalProperty` -> `@state` (we don't use TypeScript decorators) 
- [x] `static getStyles()` -> `static finalizeStyles(styles)` 
- [x] `_getUpdateComplete()` -> `getUpdateComplete()` 
- [x] `NodePart` -> `ChildPart` 

Testing checklist:  
- [x] CI/ unit tests pass 
- [x] Local testing on demo pages 
- [x] Testing component in the LMS 

For more information on the Lit 2 upgrade, visit https://lit.dev/docs/releases/upgrade/ or contact team Gaudi.

### CODE REVIEW

**Comment by lweber-d2l** [comment]
Changes look good to me. 👍 
- [ ] Coordinate Lit 2 > 3 upgrade

**Review by jorge-ramirez-arredondo — APPROVED** [review]

**Comment by jorge-ramirez-arredondo** [comment]
Close and reopen to force rerun checks.

File: package.json [hunk 1]
```diff
@@ -52,15 +52,16 @@
   },
   "dependencies": {
     "@brightspace-ui-labs/file-uploader": "^2",
```
Comment (annabelsegalld2l) [inline package.json:54]: Looks like the major version bumps for file-uploader and d2l-navigation were missed here. Would you be able to put up a PR to bump them up a major version?

Comment (lweber-d2l) [reply package.json:54]: Hi @annabelsegalld2l - I'll take care of this today.

Comment (jorge-ramirez-arredondo) [reply package.json:54]: Whoops, beat you to it Lauren. Annabel had messaged me through slack earlier. https://github.com/Brightspace/continuous-professional-development/pull/398

**Pull request merged**